packages/web-components/src/pane/pane.ts

```diff
@@ -0,0 +1,260 @@
+   * @remarks
+   * HTML Attribute: open
+   */
+  @observable
+  @attr({ mode: 'boolean' })
+  public open: boolean = false;
+
+  /**
+   * Indicates the presence of the toolbar.
+   * @public
+   * @remarks
+   * HTML Attribute: toolbar
+   */
+  @attr({ mode: 'boolean' })
+  public toolbar?: boolean = false;
+
+  /**
+   * The element to receive focus when the pane opens.
+   * @public
+   * @remarks
+   * HTML Attribute: focus-target
+   */
+  @attr({ attribute: 'focus-target' })
+  public focusTarget?: string;
+
+  /**
+   * Determines whether the focus should be trapped within the Pane when it is open.
+   * @public
+   * @remarks
+   * HTML Attribute: no-trap-focus
+   */
+  @attr({ attribute: 'trap-focus', mode: 'boolean' })
+  public trapFocus: boolean = false;
+
+  /**
+   * Sets the aria-labelledby attribute of the pane.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-labelledby
+   */
+  @attr({ attribute: 'aria-labelledby' })
+  public ariaLabelledby?: string;
+
+  /**
+   * Sets the aria-describedby attribute of the pane.
+   * @public
+   * @remarks
+   * HTML Attribute: aria-describedby
+   */
+  @attr({ attribute: 'aria-describedby' })
+  public ariaDescribedby?: string;
+
+  /**
+   * Toggles the state of the pane (open/closed).
+   * @public
+   */
+  public togglePane(): void {
+    if (!this.open && !this.hasInteracted) {
+      // Only update the hasInteracted flag if the pane is closed and hasn't been interacted with yet
+      this.hasInteracted = true;
+    }
+
+    if (!this.open) {
+      this.previousActiveElement = document.activeElement as HTMLElement;
+    } else if (this.previousActiveElement) {
+      Updates.enqueue(() => this.previousActiveElement?.focus());
+    }
+    this.open = !this.open;
+
+    this.$emit('toggled', { open: this.open, position: this.position, controlSize: this.controlSize });
+    if (this.open) this.handleFocus();
+  }
+
+  /**
+   * Shows the pane.
+   * @public
+   */
+  public openPane(): void {
+    this.previousActiveElement = document.activeElement as HTMLElement;
+    this.open = true;
+    this.$emit('opened', { open: true, position: this.position, controlSize: this.controlSize });
+    if (this.open) this.handleFocus();
+  }
+  /**
+   * Hides the pane.
+   * @public
+   */
+  public closePane(): void {
+    this.open = false;
+    this.$emit('closed', { open: false, position: this.position, controlSize: this.controlSize });
+
+    if (this.previousActiveElement) {
+      Updates.enqueue(() => this.previousActiveElement?.focus());
+    }
+  }
+
+  /**
+   * check if the item is a pane
+   */
+  private dialogRoles = ['dialog', 'complementary'];
+  private isPaneElement = (el: HTMLElement | Pane): boolean => {
+    return el instanceof Pane || (isHTMLElement(el) && this.dialogRoles.includes(el.getAttribute('role') as string));
+  };
+
+  protected setItems(): void {
+    console.log('setItems');
+    const slottedPanes = this.querySelectorAll<Pane>('fluent-pane');
```

Comment:
Why is the query on "this"? shouldn't it be this.shadowDom (or am i wrong)? Also, if using the slotted() helper in the template I don't think you need to do this? Have you tried that?